saleor/graphql/page/dataloaders.py

```diff
@@ -86,8 +76,31 @@ def map_attributes(attributes):
         )
 
 
-class AttributeValuesByPageIdLoader(DataLoader):
-    context_key = "attributevalues_by_page"
+class PageAttributesAllByPageTypeIdLoader(BasePageAttributesByPageTypeIdLoader):
+    """Loads page attributes by page type ID."""
+
+    context_key = "page_attributes_all_by_pagetype"
+
+    def get_queryset(self):
+        return AttributePage.objects.using(self.database_connection_name).all()
+
+
+class PageAttributesVisibleInStorefrontByPageTypeIdLoader(
+    BasePageAttributesByPageTypeIdLoader
+):
+    """Loads page attributes by page type ID."""
+
+    context_key = "page_attributes_visible_in_storefront_by_pagetype"
+
+    def get_queryset(self):
+        return AttributePage.objects.using(self.database_connection_name).filter(
+            attribute__visible_in_storefront=True
```

Comment:
`Exists` and `OuterRef` should be used here, as `attribute__visible_in_storefront` is generating SQL with `inner join`

Reply:
I'll open issues about these and will try to fix in the meantime.